## Análise Exploratória da Produção de Café (IBGE - Tabela 1613)
Exploração inicial dos dados processados sobre a produção de café Arábica e Canephora no Brasil, a partir da base do IBGE.

### 1 - Setup
Importação das bibliotecas e carregamento dos dados processados:

In [5]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

df = pd.read_csv('/home/brunatereza/cafe-ibge-2024/data/processed/ibge_cafe_2024_processed.csv')

### 2 - Estrutura dos Dados
Visão geral do dataset e validação inicial.

In [6]:
df.describe()
df.isna().sum()
df['UF'].value_counts()

UF
MG    494
SP    276
PR    175
BA    136
ES     75
RO     52
MT     36
AM     18
CE     18
RJ     18
AC     17
GO     15
PE      9
PA      8
MS      6
PB      2
AL      1
RN      1
TO      1
DF      1
Name: count, dtype: int64

### 3 - Completude de Valores Ausentes
Identificação de colunas com nulos e percentual de preenchimento.

In [7]:
df.isna().sum()
(df.isna().mean() * 100).round(2)


Município                                                           0.0
UF                                                                  0.0
Área destinada à colheita de Arábica                                0.0
Área destinada à colheita de Canephora                              0.0
Área colhida de Arábica                                             0.0
Área colhida de Canephora                                           0.0
Toneladas produzidas de Arábica                                     0.0
Toneladas produzidas de Canephora                                   0.0
Rendimento médio da produção (Quilogramas por Hectare) Arábica      0.0
Rendimento médio da produção (Quilogramas por Hectare) Canephora    0.0
Valor da produção (Mil Reais) Arábica                               0.0
Valor da produção (Mil Reais) Canephora                             0.0
dtype: float64

### 4- Produção Total por Estado
Quantificação da produção de Arábica e Canephora agregada por UF.

In [8]:
prod_por_uf = df.groupby('UF')[['Toneladas produzidas de Arábica', 'Toneladas produzidas de Canephora']].sum().sort_values(by='Toneladas produzidas de Arábica', ascending=False)
prod_por_uf.head(10)

Toneladas produzidas de Arábica  Toneladas produzidas de Canephora
UF                                                                    
MG                          1663992                              23337
SP                           335206                                104
ES                           224649                             657004
BA                            91890                             127260
PR                            40206                                  0
RJ                            18950                                 67
GO                            16538                                  0
DF                             1098                                  0
PE                              573                                  0
CE                              495                                 15

### 7. Conclusões da Análise Exploratória:

A análise exploratória da **Tabela 1613 (Produção Agrícola Municipal - Culturas Permanentes, IBGE 2024)** revelou padrões importantes sobre a distribuição e a produção nas diferentes regiões do país.

### Distribuição Geográfica
A base contém **1.359 municípios**, distribuídos em **20 Unidades Federativas**.  
Os estados com maior presença de municípios produtores são:

- **Minas Gerais (494 municípios)**
- **São Paulo (276 municípios)**
- **Paraná (175 municípios)**
- **Bahia (136 municípios)**
- **Espírito Santo (75 municípios)**

Esses cinco estados concentram a maior parte da produção nacional.

---

### Produção por Tipo de Café
- O **Café Arábica** domina a produção nacional, com forte presença em **MG, SP e PR**.  
- O **Café Canephora (Conilon)** aparece principalmente em **ES e BA**, refletindo a adaptação climática dessas regiões.

---

### Qualidade dos Dados
| Indicador | Valor |
|------------|--------|
| Total de linhas | 1.359 |
| Linhas com valores nulos | 0 |
| Linhas com valores zerados | 0 |
| **Completude** | **100%** |

A base de dados bruta, conta com todos os municípios do Brasil, porém muitos deles sem produção de café ou com dados omitidos, o processamento removeu todos esses registros, para a análise de registros válidos para produção de café no Brasil.
Os registros com dados válidos representam as áreas efetivamente produtoras.

---

### Conclusão
Os dados estão limpos, padronizados e prontos para análise aprofundada.  
O próximo passo será a **criação de visualizações e dashboards** para destacar:
- A concentração da produção por estado e tipo de café;  
- O valor econômico da produção;  
- As diferenças regionais de produtividade.

---

📁 *Fonte dos dados:* [IBGE – SIDRA, Tabela 1613](https://www.ibge.gov.br/estatisticas/economicas/agricultura-e-pecuaria/9117-producao-agricola-municipal-culturas-temporarias-e-permanentes)
